### PR TITLE
Update med package

### PR DESCRIPTION
**Pull request by jcortial-safran**

### CODE REVIEW

**Review by likask — APPROVED** [review]
Good change. Thank you for this.

File: var/spack/repos/builtin/packages/med/package.py [hunk 1]
```diff
@@ -16,34 +16,63 @@ class Med(CMakePackage):
...
-    depends_on("hdf5@1.10.2:1.10.7+mpi", when="@4.0.0:+mpi")
+    depends_on("hdf5@1.10.2:1.10.7+mpi", when="@4+mpi")
     depends_on("hdf5@:1.8.22~mpi", when="@3.2.0~mpi")
-    depends_on("hdf5@1.10.2:1.10.7~mpi", when="@4.0.0:~mpi")
+    depends_on("hdf5@1.10.2:1.10.7~mpi", when="@4~mpi")
     # the "TARGET hdf5" patch below only works with HDF5 shared library builds
-    depends_on("hdf5+shared", when="@4.0.0:4.1.99")
+    depends_on("hdf5+shared", when="@4")
+    depends_on("hdf5@1.12.1:1.12+mpi", when="@5:+mpi")
+    depends_on("hdf5@1.12.1:1.12~mpi", when="@5:~mpi")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/med/package.py]: Here you can group constraints differently, and simplify them:
```python
depends_on("hdf5+mpi", when="+mpi")
depends_on("hdf5~mpi", when="~mpi")
depends_on("hdf5@:1.8.22", when="@3.2.0")
depends_on("hdf5@1.10.2:1.10.7", when="@4")
depends_on("hdf5@1.12.1:1.12", when="@5:")
```
Basically, you can deal with variant activation first, and version compatibility later.

File: var/spack/repos/builtin/packages/med/package.py [hunk 2]
```diff
@@ -16,34 +16,63 @@ class Med(CMakePackage):
...
+
+    depends_on("doxygen", when="+doc")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/med/package.py]:
```suggestion
    depends_on("doxygen", type="build", when="+doc")
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a couple of minor comments, otherwise LGTM.

**Review by cessenat — APPROVED** [review]
LGTM

File: var/spack/repos/builtin/packages/med/package.py (hunk 1 above)
Comment (jcortial-safran) [reply var/spack/repos/builtin/packages/med/package.py]: Thanks for your comments, I will make the changes ASAP.

File: var/spack/repos/builtin/packages/med/package.py (hunk 2 above)
Comment (jcortial-safran) [reply var/spack/repos/builtin/packages/med/package.py]: Done

File: var/spack/repos/builtin/packages/med/package.py (hunk 1 above)
Comment (jcortial-safran) [reply var/spack/repos/builtin/packages/med/package.py]: Done

**Review by alalazo — APPROVED** [review]

**Pull request merged**